docs/source/integrations/iceberg_versioning.md

```diff
@@ -0,0 +1,228 @@
+            self.table.append(arrow)
+        else:
+            raise DatasetError("Mode not supported")
+
+    def _describe(self) -> dict:
+        return {}
+
+    def exists(self):
+        return self.catalog.table_exists((self.namespace, self.table_name))
+
+    def inspect(self):
+        return self.table.inspect
+```
+
+This dataset allows you to load Iceberg tables as `pandas`, `arrow`, `duckdb`, or `daft` objects. You can also load a subset of the table by passing the `scan_args` parameter to the dataset. The `save()` method allows you to save a dataframe as an Iceberg table. You can specify the mode of saving the table by passing the `save_args` parameter to the dataset. The `inspect()` method returns an `InspectTable` object which contains metadata about the table.
+You can also update the code to extend the functionality of the dataset to support more features of Iceberg tables. Refer to the [Iceberg API documentation](https://py.iceberg.apache.org/api/) to see what you can do with the `Table` object.
+
+## Using Iceberg tables in the catalog
+
+### Save the dataset as an Iceberg table
+
+Now update your catalog in `conf/base/catalog.yml` to update the `model_input_table` dataset to use the custom `PyIcebergDataset`:
+
+```yaml
+model_input_table:
+  type: kedro_iceberg.pyiceberg_dataset.PyIcebergDataset
+  catalog: default
+  namespace: default
+  table_name: model_input_table
+  table_type: pandas
+```
+
+Now run your Kedro project with the following command:
+
+```bash
+kedro run
+```
+You can inspect the `model_input_table` dataset created as an Iceberg table by running the following command:
+
+```bash
+find /tmp/warehouse/
+```
+The output should look something like:
+```bash
+/tmp/warehouse
+/tmp/warehouse/pyiceberg_catalog.db
+/tmp/warehouse/warehouse
+/tmp/warehouse/warehouse/default.db
+/tmp/warehouse/warehouse/default.db/model_input_table
+/tmp/warehouse/warehouse/default.db/model_input_table/data
+/tmp/warehouse/warehouse/default.db/model_input_table/data/00000-0-73afaa5b-463d-40f5-80e9-9361e93ffa82.parquet
+/tmp/warehouse/warehouse/default.db/model_input_table/metadata
+/tmp/warehouse/warehouse/default.db/model_input_table/metadata/00001-5befce5c-fc2d-4358-abd0-0b52d985fe26.metadata.json
+/tmp/warehouse/warehouse/default.db/model_input_table/metadata/snap-9089827653240705573-0-73afaa5b-463d-40f5-80e9-9361e93ffa82.avro
+/tmp/warehouse/warehouse/default.db/model_input_table/metadata/73afaa5b-463d-40f5-80e9-9361e93ffa82-m0.avro
+/tmp/warehouse/warehouse/default.db/model_input_table/metadata/00000-3dff5724-4321-419e-81fa-bb60c375bca1.metadata.json
+```
+
+Suppose the upstream datasets `companies`, `shuttles`, or `reviews` are updated. You can run the following command to generate a new version of the `model_input_table` dataset:
+
+```bash
+kedro run --to-outputs=model_input_table
+```
+You can use the `find /tmp/warehouse/` command to inspect the updated dataset and logs.
+
+### Load a specific dataset version
```

Comment:
Can you run a pipeline (and/or at least load a dataset) AS OF a certain timestamp? That would be useful for somebody who wants to use Iceberg for versioning. Not sure about loading by snapshot ID.

Reply:
Not with `pyiceberg` afaik